### PR TITLE
Sort results to make it more convenient to compare decoding results

### PR DESCRIPTION
**Pull request by pkufool**
After sorting we can compare the decoding results sequence by sequence, that would be easier to find patterns.

image

### CODE REVIEW

**Comment by pingfengluo** [comment]
One more thing, if print the WAV ID at the begin also more convenient for check the results?

**Comment by danpovey** [comment]
Looks good to me!
Eventually I think we should print some kind of file-id as well, if that's possible, but it's a separate issue.

**Comment by pkufool** [comment]
It is possible to print cut_id, cut_id usually contains file-id, I think.

**Comment by danpovey** [comment]
My instinct would be to sort by cut_id rather than transcript, then what we see when we look at the file will be more like a random sampling which can also be useful.

**Comment by pkufool** [comment]
@csukuangfj @danpovey  Please have a look at the latest commit, see whether it is ok to add cut_id this way. If so, I'll apply it to other recipes.

recogs log:
```
1089-134686-0000-459-0: ref=['HE', 'HOPED', 'THERE', 'WOULD', 'BE', 'STEW', 'FOR', 'DINNER', 'TURNIPS', 'AND', 'CARROTS', 'AND', 'BRUISED', 'POTATOES', 'AND', 'FAT', 'MUTTON', 'PIECES', 'TO', 'BE', 'LADLED', 'OUT', 'IN', 'THICK', 'PEPPERED', 'FLOUR', 'FATTENED', 'SAUCE']
1089-134686-0000-459-0: hyp=['HE', 'HOPED', 'THERE', 'WOULD', 'BE', 'STEW', 'FOR', 'DINNER', 'TURNIPS', 'AND', 'CARROTS', 'AND', 'BRUISED', 'POTATOES', 'AND', 'FAT', 'MUTTON', 'PIECES', 'TO', 'BE', 'LADLED', 'OUT', 'IN', 'THICK', 'PEPPERED', 'FLOWER', 'FAT', 'AND', 'SAUCE']
1089-134686-0001-460-0: ref=['STUFF', 'IT', 'INTO', 'YOU', 'HIS', 'BELLY', 'COUNSELLED', 'HIM']
1089-134686-0001-460-0: hyp=['STUFF', 'IT', 'INTO', 'YOU', 'HIS', 'BELLY', 'COUNSELLED', 'HIM']
1089-134686-0002-461-0: ref=['AFTER', 'EARLY', 'NIGHTFALL', 'THE', 'YELLOW', 'LAMPS', 'WOULD', 'LIGHT', 'UP', 'HERE', 'AND', 'THERE', 'THE', 'SQUALID', 'QUARTER', 'OF','THE', 'BROTHELS']
1089-134686-0002-461-0: hyp=['AFTER', 'EARLY', 'NIGHTFALL', 'THE', 'YELLOW', 'LAMPS', 'WOULD', 'LIGHT', 'UP', 'HERE', 'AND', 'THERE', 'THE', 'SQUALID', 'QUARTER', 'OF','THE', 'BROTHELS']
1089-134686-0003-462-0: ref=['HELLO', 'BERTIE', 'ANY', 'GOOD', 'IN', 'YOUR', 'MIND']
1089-134686-0003-462-0: hyp=['HELLO', 'BERTIE', 'ANY', 'GOOD', 'IN', 'YOUR', 'MIND']
1089-134686-0004-463-0: ref=['NUMBER', 'TEN', 'FRESH', 'NELLY', 'IS', 'WAITING', 'ON', 'YOU', 'GOOD', 'NIGHT', 'HUSBAND']
1089-134686-0004-463-0: hyp=['NUMBER', 'TEN', 'FRESH', 'NELLIE', 'IS', 'WAITING', 'ON', 'YOU', 'GOOD', 'NIGHT', 'HUSBAND']
```

errs log:
```
1089-134686-0000-459-0: HE HOPED THERE WOULD BE STEW FOR DINNER TURNIPS AND CARROTS AND BRUISED POTATOES AND FAT MUTTON PIECES TO BE LADLED OUT IN THICK PEPPERED (FLOUR
FATTENED->FLOWER FAT AND) SAUCE
1089-134686-0001-460-0: STUFF IT INTO YOU HIS BELLY COUNSELLED HIM
1089-134686-0002-461-0: AFTER EARLY NIGHTFALL THE YELLOW LAMPS WOULD LIGHT UP HERE AND THERE THE SQUALID QUARTER OF THE BROTHELS
1089-134686-0003-462-0: HELLO BERTIE ANY GOOD IN YOUR MIND
1089-134686-0004-463-0: NUMBER TEN FRESH (NELLY->NELLIE) IS WAITING ON YOU GOOD NIGHT HUSBAND
1089-134686-0005-464-0: THE MUSIC CAME NEARER AND HE RECALLED THE WORDS THE WORDS OF SHELLEY'S FRAGMENT UPON THE MOON WANDERING COMPANIONLESS PALE FOR WEARINESS
1089-134686-0006-465-0: THE DULL LIGHT FELL MORE FAINTLY UPON THE PAGE WHEREON ANOTHER EQUATION BEGAN TO UNFOLD ITSELF SLOWLY AND TO SPREAD ABROAD ITS WIDENING TAIL
1089-134686-0007-466-0: A COLD LUCID INDIFFERENCE REIGNED IN HIS SOUL
```

**Comment by csukuangfj** [comment]
Thanks! The output looks good to me.

**Comment by pkufool** [comment]
@csukuangfj  Please have a look at this https://github.com/k2-fsa/icefall/pull/522/commits/88400a44357f5bd42716f7001c800f7abb989192 . Do you have any suggestions?

File: egs/librispeech/ASR/pruned_transducer_stateless2/scaling.py [hunk 1]
```diff
@@ -424,18 +424,25 @@ def __init__(
...
-            return ActivationBalancerFunction.apply(
-                x,
-                self.channel_dim,
-                self.min_positive,
-                self.max_positive,
-                self.max_factor,
-                self.min_abs,
-                self.max_abs,
-            )
+            if torch.jit.is_tracing():
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless2/scaling.py:434]: Can you use
```python
if xxx:
elif yyy:
else:
```

It requires one fewer level of indent.

**Comment by pkufool** [comment]
@csukuangfj  Please have a look at the last commit, I did some other changes. Thanks!

**Comment by csukuangfj** [comment]
Thanks! It looks good to me.

**Pull request merged**